lib/super_digit.rb

```diff
@@ -1,15 +1,60 @@
 # Superdigit
-
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(log10n)?
+# Space Complexity - O(log10n)?
 def super_digit(n)
-   
+  raise ArgumentErrof if n < 0
+  return n if n < 10      # base case
+
+  sum = 0
+  while n / 10 > 0
+    sum += n % 10
+    n /= 10
+  end
+  sum += n
+  
+  return super_digit(sum)
 end
   
-
-# Time Complexity - ?
-# Space Complexity - ?
-def refined_super_digit(n, k)
-    
+# Refined Superdigit
+# Time Complexity - O(log10n)
+# Space Complexity - O(log10n)
+def refined_super_digit(n,k)
+  return super_digit(n*k)
```

Comment:
Consider a dynamic programming solution

```suggestion
def refined_super_digit(n, k)
  sum = super_digit(n)
  sum *= k
  return super_digit(sum)
```